### PR TITLE
Make menu.run a resolvable promise

### PR DESCRIPTION
**Pull request by kechkibet**

### CODE REVIEW

**Comment by kechkibet** [comment]
This will enable a person to capture ussd result especially if you wantn to save it e.g. in a db:
let ussdResult = await menu.run(req.body);

**Comment by habbes** [comment]
@kechkibet Thanks for the contribution. Could you add a test case for the changes?

**Comment by kechkibet** [comment]
Hi,
We had a situation where we had to log all menus sent to client for support purposes

File: lib/ussd-menu.js [hunk 1]
```diff
@@ -18,12 +18,16 @@ class UssdMenu extends EventEmitter {
         this.result = '';
         this.onResult = null;
         this.val = '';
+        this.resolve=null;
```
Comment (yono38) [inline lib/ussd-menu.js]: Can you add some spacing around the `=` to match the style from above please?

**Comment by yono38** [comment]
This is great @kechkibet - In addition to the test case, would you mind adding a small line to the README to note the change?

Also, we need to add a small change to the TS type definition file [here](https://github.com/habbes/ussd-menu-builder/blob/master/index.d.ts#L53) to note that it is no longer a `void` return but rather a  `Promise<string>` now

**Comment by yono38** [comment]
Thanks @kechkibet! The TS change looks good to me :D 

Would you mind making a small test case as @habbes mentioned? The codebase has very high test coverage and I think we'd like to keep it that way.

**Comment by habbes** [comment]
Thanks for the update @kechkibet, could you also mention this in the README as @yono38 suggested? And then we'll merge. Thanks.

**Comment by habbes** [comment]
Thanks @kechkibet

**Pull request merged**